### PR TITLE
Initial changes to be able to compile OMR with Clang 5.0 under Windows

### PR DESCRIPTION
**Pull request by amitin**
This is cmake-based config only.
Changes I made:
Modified cmake scripts
Removed unused variables
Fixed some type casts
Removed some {{0}} initializers, clang doesn't like it sometimes. Replaced with memset().
Made several function definitions consistent with it's declarations.

### CODE REVIEW

File: omrsigcompat/omrsig.cpp [hunk 1]
```diff
@@ -197,8 +197,12 @@ omrsig_handler(int sig, void *siginfo, void *uc)
 sighandler_t
 omrsig_primary_signal(int signum, sighandler_t handler)
 {
-	struct sigaction act = {{0}};
-	struct sigaction oldact = {{0}};
+	struct sigaction act;
+	struct sigaction oldact;
+
+	memset(&act, 0, sizeof(struct sigaction));
```
Comment (charliegracie) [inline omrsigcompat/omrsig.cpp]: The zero initializer gives a warning on clang?

File: port/win32/omrfile_blockingasync.c [hunk 2]
```diff
@@ -139,7 +139,7 @@ omrfile_blockingasync_read(struct OMRPortLibrary *portLibrary, intptr_t fd, void
 
 
 intptr_t
-omrfile_blockingasync_write(struct OMRPortLibrary *portLibrary, intptr_t fd, void *buf, intptr_t nbytes)
+omrfile_blockingasync_write(struct OMRPortLibrary *portLibrary, intptr_t fd, const void *buf, intptr_t nbytes)
```
Comment (charliegracie) [inline port/win32/omrfile_blockingasync.c]: Can you update all instances of `omrfile_blockingasync_write` to have the proper signature?  I see a few other hits in the codebase missing the `const`.  Thanks

File: port/win32/omrintrospect.c [hunk 3]
```diff
@@ -135,8 +136,7 @@ suspend_all_preemptive(struct PlatformWalkData *data)
 		}
 
 		if (!Thread32First(data->snapshot, &data->thread32)) {
-			result = GetLastError();
-			goto error_return;
```
Comment (charliegracie) [inline port/win32/omrintrospect.c]: Can you provide a bit more context why you are changing the behaviour here?

File: port/win32/omrintrospect.c [hunk 4]
```diff
@@ -176,13 +175,8 @@ suspend_all_preemptive(struct PlatformWalkData *data)
 		passA = !passA;
 	} while (a_count != b_count);
 
-	if (a_count != b_count) {
-		goto error_return;
```
Comment (charliegracie) [inline port/win32/omrintrospect.c]: Why are you removing this?

File: omrsigcompat/omrsig.cpp (hunk 1 above)
Comment (amitin) [reply omrsigcompat/omrsig.cpp]: Yes, it does.

File: port/win32/omrfile_blockingasync.c (hunk 2 above)
Comment (amitin) [reply port/win32/omrfile_blockingasync.c]: Sure.

File: port/win32/omrintrospect.c (hunk 3 above)
Comment (amitin) [reply port/win32/omrintrospect.c]: The code at 'error_return' label does SetLastError(result) and returns -1. So doing GetLastError() and then immediately invoking SetLastError() seems to be a noop. Why don't just return -1 having last error set by Thread32First() function?
See also the code above: CreateToolhelp32Snapshot() also sets last error if it fails, and the check against invalid handle just returns -1 - the last error is set.

File: port/win32/omrintrospect.c (hunk 4 above)
Comment (amitin) [reply port/win32/omrintrospect.c]: This condition is always false. The execution flow only gets there if the loop above exits normally.
Additionally, if the condition somehow wasn't false, then there is a chance to get to 'error_return' label having 'result' variable uninitialised.

**Comment by amitin** [comment]
Closing this. See #2198.

**Pull request closed without merging**